src/client/main.c

```diff
@@ -10,42 +10,31 @@
 
 const struct option options[] = { { ARG_HELP, no_argument, 0, ARG_HELP_SHORT }, { NULL, 0, 0, '\0' } };
 
-#define OPTIONS_STR ARG_HELP_SHORT_S
+#define OPTIONS_STR ARG_HELP_SHORT_S ARG_FILTER_SHORT_S
 
 static char *op;
 static char **opargv;
 static int opargc;
 static bool no_action = false;
 
-static void usage(char *argv[]) {
-        print_client_usage(argv[0]);
-}
-
 static int get_opts(int argc, char *argv[]) {
-        int opt = 0;
-
-        while ((opt = getopt_long(argc, argv, OPTIONS_STR, options, NULL)) != -1) {
-                if (opt == ARG_HELP_SHORT) {
-                        usage(argv);
-                        no_action = true;
-                        return 0;
-                } else {
-                        fprintf(stderr, "Unsupported option %c\n", opt);
-                        usage(argv);
-                        return -EINVAL;
-                }
-        }
-
-        if (optind < argc) {
-                op = argv[optind++];
-                opargv = &argv[optind];
-                opargc = argc - optind;
-        } else {
+        if (argc == 1) {
                 fprintf(stderr, "No command given\n");
-                usage(argv);
+                print_client_usage(argv[0]);
                 return -EINVAL;
         }
 
+        op = argv[1];
+        if (streq(op, ARG_HELP) || streq(op, "--" ARG_HELP) || streq(op, "-h")) {
```

Comment:
> Usually specifying --help anywhere will lead to printing the help. In various tools it even allows to print separate help for different commands (though we are not yet there).

The way it works now is that `hirtectl --help` will print the general usage of the tool - which commands are there and what are they used for. This strictly refers to the first parameter given. So the cmd `hirtectl list-units --help` will print the detailed usage of the `list-units` command. As stated above, I am not happy about placing the option parsing in the `client.c`, but didn't want to introduce some kind of command builder (yet) as that would also require refactoring all other commands - which seemed like a pretty big change for one PR. 

> I think the existing infra should be preserved, and --filter flag should be processed here in main.c and passed over to further handling. Preferably as a Printer attribute

Same argument as above, the `--filter` option is specific to the `list-units` command. Placing it on the printer struct makes it available to all the other prints.

Reply:
> Same argument as above, the --filter option is specific to the list-units command. Placing it on the printer struct makes it available to all the other prints.

It can be either ignored or failed... But parsing command line args this way is very ugly.